app/Console/Commands/ImportGreenlight.php

```diff
@@ -0,0 +1,323 @@
+            'prefix_indexes' => true,
+            'schema'         => 'public',
+            'sslmode'        => 'prefer',
+        ]);
+
+        $requireAuth    = DB::connection('greenlight')->table('features')->where('name', 'Room Authentication')->first('value')->value;
+        $users          = DB::connection('greenlight')->table('users')->where('deleted', false)->get(['id','provider','username','email','name','password_digest']);
+        $rooms          = DB::connection('greenlight')->table('rooms')->where('deleted', false)->get(['id','uid','user_id','name','room_settings','access_code']);
+        $sharedAccesses = DB::connection('greenlight')->table('shared_accesses')->get(['room_id','user_id']);
+
+        // ask user what room type the imported rooms should get
+        $roomTypeShort = $this->choice(
+            'What room type should the rooms be assigned to?',
+            RoomType::all()->pluck('short')->toArray(),
+            0,
+            $maxAttempts = null,
+            $allowMultipleSelections = false
+        );
+
+        // ask user to add prefix to room names
+        $prefix = $this->ask('Prefix for room names:');
+
+        // find id of the selected roomType
+        $roomType = RoomType::where('short', $roomTypeShort)->first()->id;
+
+        $userMap = $this->importUsers($users);
+        $roomMap = $this->importRooms($rooms, $roomType, $userMap, !$requireAuth, $prefix);
+        $this->importSharedAccesses($sharedAccesses, $roomMap, $userMap);
+    }
+
+    /**
+     * Process greenlight user collection and try to import users
+     *
+     * @param  Collection $users Collection with all users found in the greenlight database
+     * @return array      Array map of greenlight user ids as key and id of the found/created user as value
+     */
+    protected function importUsers(Collection $users): array
+    {
+        $this->line('Importing users');
+        $userMap  = [];
+
+        $bar = $this->output->createProgressBar($users->count());
+        $bar->start();
+
+        // counter of user ids that already exists
+        $existed = 0;
+        // counter of users that are created
+        $created = 0;
+        // list of user that could not be created, e.g. ldap import failed
+        $failed = [];
+
+        foreach ($users as $user) {
+
+            // import greenlight users
+            if ($user->provider == 'greenlight') {
+
+                // check if user with this email exists
+                $dbUser = User::where('email', $user->email)->where('authenticator', 'users')->first();
+                if ($dbUser != null) {
+                    // user found, link greenlight user id to id of found user
+                    $existed++;
+                    $userMap[$user->id] = $dbUser->id;
+                    $bar->advance();
+
+                    continue;
+                }
+
+                // create new user
+                $dbUser                = new User();
```

Comment:
Shouldn't we add a selectable default role for greenlight db users?

Reply:
Good idea, or a multiselect

Reply:
I think I also forgot the timezone default

Reply:
As laravel does not support muliselects in tests, I only gave the option to select one role